tachyon/math/finite_fields/prime_field_base.h

```diff
@@ -165,11 +173,12 @@ class Copyable<
   static bool s_is_in_montgomery;
 
   static bool WriteTo(const T& prime_field, Buffer* buffer) {
-    if (s_is_in_montgomery) {
-      return buffer->Write(prime_field.ToMontgomery());
-    } else {
-      return buffer->Write(prime_field.ToBigInt());
+    if constexpr (T::Config::kUseMontgomery) {
+      if (s_is_in_montgomery) {
+        return buffer->Write(prime_field.value());
+      }
```

Comment:
```suggestion
    if constexpr (T::Config::kUseMontgomery && s_is_in_montgomery) {
      return buffer->Write(prime_field.value());
  ```
  Also change for the other 3 similar occurrences in this file

Reply:
We can't do this. since `s_is_in_montgomery` is not determined at compile time.